misc/generate_gitrev.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+GITREV=$(git rev-parse --short HEAD)
+GITTAG=$(git tag --points-at ${GITREV})
```

Comment:
Do we need error checks (e.g. `git` command returning an error)?

I wouldn't mind if you wrote this script using Perl, if use of sh makes you negative to having error checks.